src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -945,7 +956,8 @@ VideoDecoder::DecodedOutput VideoDecoder::getFrameDisplayedAtTimestampNoDemux(
         return seconds >= frameStartTime && seconds < frameEndTime;
       });
   // Convert the frame to tensor.
-  return convertAVFrameToDecodedOutput(rawOutput);
+  auto preAllocatedOutputTensor = torch::empty({0});
```

Comment:
We are now creating "fake pre-allocated tensors" in a bunch of places. Do I extract that out in a `Decoder::MakeDummyPreAllocatedTensor()` method to clarify the meaning behind all this?

Reply:
Why do we need to create these tensors? I thought having an optional would allow us to only pass this in when we actually need it.

Reply:
See https://github.com/pytorch/torchcodec/pull/266/files#r1803042354 for why I wasn't able to use optional here